R/zchunk_L201.en_nonco2.R

```diff
@@ -0,0 +1,358 @@
+      na.omit %>%
+      # No need to include a GDP control when the max.reduction is zero
+      filter(max_reduction > 0) ->
+      L201.nonghg_gdp_control_res
+
+    L201.nonghg_max_reduction_res <- select(L201.nonghg_gdp_control_res, -steepness)
+    L201.nonghg_steepness_res <- select(L201.nonghg_gdp_control_res, -max_reduction)
+
+    # Rename to regional SO2
+    L201.en_pol_emissions <- rename_SO2(L201.en_pol_emissions, A_regions, FALSE)
+    L201.nonghg_max_reduction <- rename_SO2(L201.nonghg_max_reduction, A_regions, FALSE)
+    L201.nonghg_steepness <- rename_SO2(L201.nonghg_steepness, A_regions, FALSE)
+    L201.nonghg_res <- rename_SO2(L201.nonghg_res, A_regions, FALSE)
+    L201.nonghg_steepness_res <- rename_SO2(L201.nonghg_steepness_res, A_regions, FALSE)
+    L201.nonghg_max_reduction_res <- rename_SO2(L201.nonghg_max_reduction_res, A_regions, FALSE)
+
+    # Remove district heat from regions that do have have it
+    A_regions.en %>%
+      filter(has_district_heat == 1) %>%
+      .[["region"]] ->
+      distheat.regions
+    L201.en_pol_emissions <- filter(L201.en_pol_emissions, supplysector != "district heat" | region %in% distheat.regions)
+    L201.en_ghg_emissions <- filter(L201.en_ghg_emissions, supplysector != "district heat" | region %in% distheat.regions)
+    L201.en_bcoc_emissions <- filter(L201.en_bcoc_emissions, supplysector != "district heat" | region %in% distheat.regions)
+    L201.nonghg_max_reduction <- filter(L201.nonghg_max_reduction, supplysector != "district heat" | region %in% distheat.regions)
+    L201.nonghg_steepness <- filter(L201.nonghg_steepness, supplysector != "district heat" | region %in% distheat.regions)
+
+    # It may be the case with certain regional aggregations that regions exist that have no
+    # heating or cooling sectors. We should delete those here.
+    # Delete sectors that do not exist due to zero heating/cooling degree days
+    delete_zerodays <- function(x, L201.delete.sectors) {
+      filter(x, ! paste0(region, supplysector) %in% L201.delete.sectors)
+    }
+    L201.delete.sectors <- paste0(L244.DeleteThermalService$region, L244.DeleteThermalService$supplysector )
+    L201.en_pol_emissions <- delete_zerodays(L201.en_pol_emissions, L201.delete.sectors)
+    L201.en_ghg_emissions <- delete_zerodays(L201.en_ghg_emissions, L201.delete.sectors)
+    L201.en_bcoc_emissions <- delete_zerodays(L201.en_bcoc_emissions, L201.delete.sectors)
+    L201.nonghg_max_reduction <- delete_zerodays(L201.nonghg_max_reduction, L201.delete.sectors)
+    L201.nonghg_steepness <- delete_zerodays(L201.nonghg_steepness, L201.delete.sectors)
+
+
+    # Produce outputs
+    L201.en_pol_emissions %>%
+      add_title("Pollutant emissions for energy technologies in all regions") %>%
+      add_units("Tg") %>%
+      add_comments("Take non-GHG emissions for the energy system, filter to out_resources") %>%
```

Comment:
I think you are filtering out resources, not leaving them in. This is because resource emissions have to be read into GCAM differently than technology emissions.